tests/cli/test_traces.py

```diff
@@ -0,0 +1,251 @@
+    mock_result.token = None
+
+    # Patch the TracingClient at the module level where it's imported
+    with mock.patch("mlflow.cli.traces.TracingClient") as mock_client:
+        mock_instance = mock.Mock()
+        mock_client.return_value = mock_instance
+        mock_instance.search_traces.return_value = mock_result
+
+        result = runner.invoke(
+            commands,
+            ["search", "--experiment-id", "1", "--extract-fields", "info.trace_id,info.state"],
+        )
+
+        assert result.exit_code == 0
+        # Check that either the table output or the values appear in output
+        assert "tr-123" in result.output
+        assert "OK" in result.output
+
+
+def test_get_command_with_fields(runner):
+    """Test get command with field selection."""
+    mock_trace = {
+        "info": {"trace_id": "tr-123", "state": "OK"},
+        "data": {"spans": [{"name": "test_span"}]},
+    }
+
+    # Create a mock trace object with a to_dict method
+    mock_trace_obj = mock.Mock()
+    mock_trace_obj.to_dict.return_value = mock_trace
+
+    # Patch the TracingClient at the module level where it's imported
+    with mock.patch("mlflow.cli.traces.TracingClient") as mock_client:
+        mock_instance = mock.Mock()
+        mock_client.return_value = mock_instance
+        mock_instance.get_trace.return_value = mock_trace_obj
+
+        result = runner.invoke(
+            commands,
+            ["get", "--trace-id", "tr-123", "--extract-fields", "info.trace_id"],
+        )
+
+        assert result.exit_code == 0
+
+        # Parse JSON output
+        output_json = json.loads(result.output)
+        assert output_json == {"info": {"trace_id": "tr-123"}}
+
+
+def test_delete_command(runner):
+    """Test delete command."""
+    # Patch the TracingClient at the module level where it's imported
+    with mock.patch("mlflow.cli.traces.TracingClient") as mock_client:
+        mock_instance = mock.Mock()
+        mock_client.return_value = mock_instance
+        mock_instance.delete_traces.return_value = 5
+
+        result = runner.invoke(
+            commands,
+            ["delete", "--experiment-id", "1", "--trace-ids", "tr-1,tr-2,tr-3"],
+        )
+
+        assert result.exit_code == 0
+        assert "Deleted 5 trace(s)" in result.output
+
+
+def test_field_validation_error(runner):
+    """Test that invalid fields produce helpful error messages."""
+    mock_trace = {"info": {"trace_id": "tr-123"}}
+
+    # Create a mock trace object with a to_dict method
+    mock_trace_obj = mock.Mock()
+    mock_trace_obj.to_dict.return_value = mock_trace
+
+    # Create a mock result that acts like a list but also has a token attribute
+    mock_result = mock.Mock()
+    mock_result.__iter__ = lambda self: iter([mock_trace_obj])
+    mock_result.__getitem__ = lambda self, i: mock_trace_obj if i == 0 else None
+    mock_result.__len__ = lambda self: 1
+    mock_result.__bool__ = lambda self: True
+    mock_result.token = None
+
```

Comment:
can we replace this with:


https://github.com/mlflow/mlflow/blob/7f6ce43394679c4285942308cd8dd86bc6e55856/mlflow/store/entities/paged_list.py#L6